packages/client/src/components/app/Button.svelte

```diff
@@ -16,6 +18,16 @@
   export let icon = null
   export let active = false
 
+  const handleOnClick = async () => {
+    if (handlingOnClick) {
```

Comment:
Instead of returning here, could we not use the `handlingOnClick` flag to disable the button instead? Then you also get visual feedback that something is happening, and it won't fire any more onclick events while disabled. I'm fairly certain all you need to do is pass in the actual `disabled` attribute and spectrum CSS should have styles for it.